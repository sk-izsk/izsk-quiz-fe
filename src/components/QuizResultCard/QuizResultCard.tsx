import { Box, makeStyles } from '@material-ui/core';
import React, { useState } from 'react';
import { Body2, Button, Card, CardAction, CardContent, H5, Subtitle2 } from 'ui-neumorphism';
import { QuizResultModal } from '..';
import { CustomTheme, theme } from '../../theme/muiTheme';

export interface QuizResultCardProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  cardContainer: {
    padding: theme.spacing(1.5),
    maxWidth: 300,
    width: '100%',
    cursor: 'pointer',
  },
  card: {
    padding: theme.spacing(1),
  },
  dateContainer: {
    marginTop: theme.spacing(1.5),
    marginBottom: theme.spacing(1.5),
  },
}));

const QuizResultCard: React.FC<QuizResultCardProps> = () => {
  const classes = useStyles();
  const [inset, setInset] = useState<boolean>(false);
  const [openModal, setOpenModal] = useState<boolean>(false);
  const handleInset = () => setInset(!inset);
  const handleCloseModal = () => setOpenModal(false);
  const handleOpenModal = () => setOpenModal(true);
  return (
    <Box className={classes.cardContainer} onMouseOver={handleInset} onMouseOut={handleInset}>
      <QuizResultModal visible={openModal} onClose={handleCloseModal} />
      <Card className={classes.card} bordered inset={inset}>
        <CardContent>
          <H5>Quiz Title</H5>
          <Subtitle2 className={classes.dateContainer} secondary>
            date of quiz
          </Subtitle2>
          <Body2>
            Result of quiz
            <br />
            Passed
          </Body2>
        </CardContent>
        <CardAction>
          <Button
            onClick={handleOpenModal}
            rounded
            bgColor={theme.palette.primary.main}
            color={theme.palette.success.contrastText}
          >
            More details
          </Button>
        </CardAction>
      </Card>
    </Box>
  );
};

export { QuizResultCard };
