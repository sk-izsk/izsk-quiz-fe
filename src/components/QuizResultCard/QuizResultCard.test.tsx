import { render } from '@testing-library/react';
import React from 'react';
import { QuizResultCard } from './QuizResultCard';

test('QuizResultCard renders correctly', () => {
  const { asFragment } = render(<QuizResultCard />);
  expect(asFragment()).toMatchSnapshot();
});
