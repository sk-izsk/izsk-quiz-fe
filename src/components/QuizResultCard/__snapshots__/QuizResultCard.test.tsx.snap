// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizResultCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-cardContainer-1"
  >
    <div
      class="_28vIk _2bGnb _1SFBz _12WTq makeStyles-card-16"
    >
      <div
        class="_2So3v _2uw1B "
      >
        <h5
          class="_3yPwv _2TwOi _11JjB _1uvOH "
        >
          Quiz Title
        </h5>
        <p
          class="_3yPwv _2vNlX _1QRSN _1uvOH makeStyles-dateContainer-3"
        >
          date of quiz
        </p>
        <p
          class="_3yPwv _i_J81 _11JjB _1uvOH "
        >
          Result of quiz
          <br />
          Passed
        </p>
      </div>
      <div
        class="_qCt7a _1l5Ig "
      >
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND "
          style="color: rgb(238, 252, 243); background-color: rgb(33, 150, 243);"
        >
          <button
            class="_3lLt8"
          >
            More details
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
